contracts/MANACrowdsale.sol

```diff
@@ -42,17 +45,29 @@ contract MANACrowdsale is ContinuousCrowdsale, CappedCrowdsale, WhitelistedCrowd
         return new MANAToken();
     }
 
-    function getRate() internal returns(uint256) {
-        // whitelisted investors can purchase at preferential price before crowdsale ends
-        if (isWhitelisted(msg.sender) && !hasEnded()) {
-            return preferentialRate;
-        }
+    function setBuyerRate(address beneficiary, uint256 rate) onlyOwner {
+        require(beneficiary != 0);
+        require(rate != 0);
+
+        buyerRate[beneficiary] = rate;
+    }
```

Comment:
That clears things up. I was worried that the `beneficiary` / `investor` split would force us to redesign `validPurchase` to accept a beneficiary.